spec/models/tag_spec.rb

```diff
@@ -261,6 +261,27 @@ def previous_name_error_message
     end
   end
 
+  describe '.find_or_create_by_names_race_condition' do
+    it 'handles simultaneous inserts of the same tag in different cases without error' do
+      tag_name_upper = 'Rails'
+      tag_name_lower = 'rails'
+
+      threads = []
+
+      2.times do |i|
+        threads << Thread.new do
+          Tag.find_or_create_by_names(i.zero? ? tag_name_upper : tag_name_lower)
```

Comment:
The linter is complaining about the following:
```suggestion
          described_class.find_or_create_by_names(i.zero? ? tag_name_upper : tag_name_lower)
```